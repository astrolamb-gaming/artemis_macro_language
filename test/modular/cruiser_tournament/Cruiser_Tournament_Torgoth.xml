<mission_data>
  <imports>
    <import name="Cruiser_Tournament_templates.xml" />
  </imports>
  <values>
    <range name="torgoth">
      <struct>
        <value name="Behemoth 1" x="22222.0" y="-500.0" z="22200.0" size="large" hull="Behemoth" />
        <value weight="150" surrender="75" ship="0.0" captain="1.0" />
      </struct>
      <struct>
        <value name="Goliath 1" x="22000.0" y="0.0" z="23000.0" size="small" hull="Goliath" />
        <value weight="80" surrender="40" ship="1.0" captain="-1.0" />
      </struct>
      <struct>
        <value name="Goliath 2" x="22433.0" y="500.0" z="21800.0" size="small" hull="Goliath" />
        <value weight="80" surrender="40" ship="-1.0" captain="-1.0" />
      </struct>
      <struct>
        <value name="Leviathan 1" x="22000.0" y="0.0" z="23000.0" size="medium" hull="Leviathan" />
        <value weight="90" surrender="45" ship="0.0" captain="0.0" />
      </struct>
      <struct>
        <value name="Leviathan 2" x="21000.0" y="0.0" z="22000.0" size="medium" hull="Leviathan" />
        <value weight="90" surrender="45" ship="-1.0" captain="-1.0" />
      </struct>
    </range>
  </values>
  <start>
    <repeat _range="torgoth">
      <create type="enemy" x="${x}" y="${y}" z="${z}" angle="45" name="${name}" raceKeys="Torgoth enemy" hulKeysl="${size} ${hull}" size="" fleetnumber="5"/>
      <set_special name="${name}" ship="0.0" captain="1.0"/>
    </repeat>
  </start>
  <repeat _range="torgoth">
    <!--Override text to match original  -->
    <expand _template="tonnage" race="Enemy" hull="Torgoth"  />
  </repeat>
</mission_data>