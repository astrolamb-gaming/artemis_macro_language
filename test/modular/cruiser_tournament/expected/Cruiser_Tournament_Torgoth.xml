<mission_data>
  <start>
    <create type="enemy" x="22222.0" y="-500.0" z="22200.0" angle="45" name="Behemoth 1" raceKeys="Torgoth enemy" hullKeys="Behemoth large" fleetnumber="5"/>
    <set_special name="Behemoth 1" ship="0.0" captain="1.0"/>
    <create type="enemy" x="22000.0" y="0.0" z="23000.0" angle="45" name="Goliath 1" raceKeys="Torgoth enemy" hullKeys="Goliath small" fleetnumber="5"/>
    <set_special name="Goliath 1" ship="1.0" captain="-1.0"/>
    <create type="enemy" x="22433.0" y="500.0" z="21800.0" angle="45" name="Goliath 2" raceKeys="Torgoth enemy" hullKeys="Goliath small" fleetnumber="5"/>
    <set_special name="Goliath 2" ship="-1.0" captain="-1.0"/>
    <create type="enemy" x="22000.0" y="0.0" z="23000.0" angle="45" name="Leviathan 1" raceKeys="Torgoth enemy" hullKeys="Leviathan medium" fleetnumber="5"/>
    <set_special name="Leviathan 1" ship="0.0" captain="0.0"/>
    <create type="enemy" x="21000.0" y="0.0" z="22000.0" angle="45" name="Leviathan 2" raceKeys="Torgoth enemy" hullKeys="Leviathan medium" fleetnumber="5"/>
    <set_special name="Leviathan 2" ship="-1.0" captain="-1.0"/>
  </start>
  <event name="Behemoth 1 Surrendered">
    <if_variable name="Behemoth 1_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Behemoth 1" comparator="EQUALS" value="1"/>
    <set_variable name="Behemoth 1_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Behemoth 1"/>
    <set_variable name="Tonnage" value="Tonnage+150+75"/>
    <big_message title="Torgoth Behemoth surrendered" subtitle1="150 kilotons" subtitle2="+75 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Behemoth Behemoth 1 Surrendered"/>
  </event>
  <event name="Behemoth 1 Destroyed">
    <if_variable name="Behemoth 1_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Behemoth 1_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Behemoth 1"/>
    <set_variable name="Behemoth 1_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage+150"/>
    <big_message title="Torgoth Behemoth Behemoth 1 destroyed" subtitle1="150 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Behemoth Behemoth 1 destroyed"/>
  </event>
  <event name="Goliath 1 Surrendered">
    <if_variable name="Goliath 1_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Goliath 1" comparator="EQUALS" value="1"/>
    <set_variable name="Goliath 1_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Goliath 1"/>
    <set_variable name="Tonnage" value="Tonnage+80+40"/>
    <big_message title="Torgoth Goliath surrendered" subtitle1="80 kilotons" subtitle2="+40 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Goliath Goliath 1 Surrendered"/>
  </event>
  <event name="Goliath 1 Destroyed">
    <if_variable name="Goliath 1_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Goliath 1_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Goliath 1"/>
    <set_variable name="Goliath 1_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage+80"/>
    <big_message title="Torgoth Goliath Goliath 1 destroyed" subtitle1="80 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Goliath Goliath 1 destroyed"/>
  </event>
  <event name="Goliath 2 Surrendered">
    <if_variable name="Goliath 2_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Goliath 2" comparator="EQUALS" value="1"/>
    <set_variable name="Goliath 2_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Goliath 2"/>
    <set_variable name="Tonnage" value="Tonnage+80+40"/>
    <big_message title="Torgoth Goliath surrendered" subtitle1="80 kilotons" subtitle2="+40 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Goliath Goliath 2 Surrendered"/>
  </event>
  <event name="Goliath 2 Destroyed">
    <if_variable name="Goliath 2_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Goliath 2_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Goliath 2"/>
    <set_variable name="Goliath 2_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage+80"/>
    <big_message title="Torgoth Goliath Goliath 2 destroyed" subtitle1="80 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Goliath Goliath 2 destroyed"/>
  </event>
  <event name="Leviathan 1 Surrendered">
    <if_variable name="Leviathan 1_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Leviathan 1" comparator="EQUALS" value="1"/>
    <set_variable name="Leviathan 1_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Leviathan 1"/>
    <set_variable name="Tonnage" value="Tonnage+90+45"/>
    <big_message title="Torgoth Leviathan surrendered" subtitle1="90 kilotons" subtitle2="+45 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Leviathan Leviathan 1 Surrendered"/>
  </event>
  <event name="Leviathan 1 Destroyed">
    <if_variable name="Leviathan 1_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Leviathan 1_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Leviathan 1"/>
    <set_variable name="Leviathan 1_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage+90"/>
    <big_message title="Torgoth Leviathan Leviathan 1 destroyed" subtitle1="90 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Leviathan Leviathan 1 destroyed"/>
  </event>
  <event name="Leviathan 2 Surrendered">
    <if_variable name="Leviathan 2_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Leviathan 2" comparator="EQUALS" value="1"/>
    <set_variable name="Leviathan 2_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Leviathan 2"/>
    <set_variable name="Tonnage" value="Tonnage+90+45"/>
    <big_message title="Torgoth Leviathan surrendered" subtitle1="90 kilotons" subtitle2="+45 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Leviathan Leviathan 2 Surrendered"/>
  </event>
  <event name="Leviathan 2 Destroyed">
    <if_variable name="Leviathan 2_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Leviathan 2_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Leviathan 2"/>
    <set_variable name="Leviathan 2_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage+90"/>
    <big_message title="Torgoth Leviathan Leviathan 2 destroyed" subtitle1="90 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Torgoth Leviathan Leviathan 2 destroyed"/>
  </event>
</mission_data>